### ISSUE TITLE
Display images with higher resolution

### ISSUE THREAD

**Issue by mcastorina** [zim-desktop-wiki/zim-desktop-wiki#993]
Screenshots that I paste into zim are always displayed very large (e.g. 1920x1280), but when I scale down the size I lose resolution and the image becomes blurry.

It would be great to be able to modify the resolution so we maintain image details but display them in a smaller area.

**Comment by dehesselle** [comment]
Right-click on the pasted screenshot, "Edit Properties" and scale it down there. The original remains untouched.

**Comment by mcastorina** [comment]
Yes, I have tried that. When I scale it down using the "Edit Properties" window, the resulting image is blurry.

**Comment by dehesselle** [comment]
I can't reproduce that.
You mean the scaled down image in Zim looks different from when you scale it down with some other program yourself?

**Comment by mcastorina** [comment]
Yes, the scaled down image looks blurry than when I view it in another application. Maybe I'm not finding the right words.. here is a screenshot of a scaled down image in Zim.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/5395814/71709649-21a94900-2dbe-11ea-9be6-9b015118e960.png">

**Original Image**
![image](https://user-images.githubusercontent.com/5395814/71709796-c3c93100-2dbe-11ea-80aa-c86e0b26ff30.png)

The problem I'm finding is that the original is displayed too big in Zim, so I have to scale it down, and when I do the result looks blurred.

**Comment by Self-Perfection** [comment]
@mcastorina do you have HiDPI screen? Looks like HiDPI issue to me.

**Comment by mcastorina** [comment]
I believe so. This is on a 15.4 inch MacBook (2880 x 1800)

**Comment by mcastorina** [comment]
I dug into the code and I think this is the relevant line: https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/3bb7910322eb6331a5cbf709d6a44a245665a7cd/zim/gui/pageview.py#L1120

It looks to me that this is simply a limitation of Gtk's image scaling. I haven't worked with Gtk before, so please correct me if I am wrong, but there is no way to set the resolution of the image widget.

Also, what is confusing me is if a screenshot is 2000x1000 px, and my monitor is 2000 pixels wide, shouldn't the image fit on the screen? Why does Gtk blow it up so large?

**Comment by jaap-karssenberg** [comment]
In current develop branch this line now allows selecting scaling algorithm used
https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/382189a9a1a68b8c6173e5a71019403dd3b492d5/zim/gui/pageview/__init__.py#L507

Maybe worth while playing with the scaling options to see if that gives better results

**Comment by sorrge** [comment]
I have the same problem in ver. 0.74.3. It is a HiDPI issue. I use XFCE with 2x upscaling. All fonts are displayed correctly, but the images are 2x upscaled and thus blurry. Downsizing them doesn't remove the blur, because it is the display function that upscales them. So if I downsize the image in the size dialog, it is first downscaled and then upscaled again on display, resulting in blur. I don't believe that different interpolation is going to solve the problem. Somehow the function that displays the image needs to be HiDPI-aware and remove the upscaling.
